include/sound/sof/header2.h

```diff
@@ -0,0 +1,410 @@
+	DEPTH_24BIT	= 24, /* 24 bits depth - Default */
+	DEPTH_32BIT	= 32, /* 32 bits depth */
+	DEPTH_64BIT	= 64, /* 64 bits depth */
+	DEPTH_INVALID
+};
+
+enum channel_config {
+	/* one channel only. */
+	CHANNEL_CONFIG_MONO			= 0,
+	/* L & R. */
+	CHANNEL_CONFIG_STEREO			= 1,
+	/* L, R & LFE; PCM only. */
+	CHANNEL_CONFIG_2_POINT_1		= 2,
+	/* L, C & R; MP3 & AAC only. */
+	CHANNEL_CONFIG_3_POINT_0		= 3,
+	/* L, C, R & LFE; PCM only. */
+	CHANNEL_CONFIG_3_POINT_1		= 4,
+	/* L, R, Ls & Rs; PCM only. */
+	CHANNEL_CONFIG_QUATRO			= 5,
+	/* L, C, R & Cs; MP3 & AAC only. */
+	CHANNEL_CONFIG_4_POINT_0		= 6,
+	/* L, C, R, Ls & Rs. */
+	CHANNEL_CONFIG_5_POINT_0		= 7,
+	/* L, C, R, Ls, Rs & LFE. */
+	CHANNEL_CONFIG_5_POINT_1		= 8,
+	/* one channel replicated in two. */
+	CHANNEL_CONFIG_DUAL_MONO		= 9,
+	/* Stereo (L,R) in 4 slots, 1st stream: [ L, R, -, - ] */
+	CHANNEL_CONFIG_I2S_DUAL_STEREO_0	= 10,
+	/* Stereo (L,R) in 4 slots, 2nd stream: [ -, -, L, R ] */
+	CHANNEL_CONFIG_I2S_DUAL_STEREO_1	= 11,
+	/* L, C, R, Ls, Rs & LFE., LS, RS */
+	CHANNEL_CONFIG_7_POINT_1		= 12,
+	CHANNEL_CONFIG_INVALID
+};
+
+enum interleaved_style {
+	CHANNELS_INTERLEAVED = 0,
+	CHANNELS_NONINTERLEAVED = 1,
+};
+
+enum sample_type {
+	MSB_INTEGER = 0, /* integer with Most Significant Byte first */
+	LSB_INTEGER = 1, /* integer with Least Significant Byte first */
+	SIGNED_INTEGER = 2, /* signed integer */
+	UNSIGNED_INTEGER = 3, /* unsigned integer */
+	FLOAT = 4 /* unsigned integer */
+};
+
+struct audio_format {
+	enum sampling_frequency sampling_frequency;
+	enum bit_depth bit_depth;
+	uint32_t ch_map;
+	enum channel_config ch_cfg;
+	uint32_t interleaving_style;
+	uint32_t channels_count : 8;
+	uint32_t valid_bit_depth : 8;
+	enum sample_type s_type : 8;
+	uint32_t reserved : 8;
+};
+
+struct basic_module_cfg {
+	uint32_t cpc; /* the max count of Cycles Per Chunk processing */
+	uint32_t ibs; /* input Buffer Size (in bytes)  */
+	uint32_t obs; /* output Buffer Size (in bytes) */
+	uint32_t is_pages; /* number of physical pages used */
+	struct audio_format audio_fmt;
+};
+
+enum node_type {
+	/* HD/A host output (-> DSP) */
+	kHdaHostOutputClass = 0,
+	/* HD/A host input (<- DSP) */
+	kHdaHostInputClass = 1,
+	/* HD/A host input/output (rsvd for future use) */
+	kHdaHostInoutClass = 2,
+
+	/* HD/A link output (DSP ->) */
+	kHdaLinkOutputClass = 8,
+	/* HD/A link input (DSP <-) */
+	kHdaLinkInputClass = 9,
+	/* HD/A link input/output (rsvd for future use) */
+	kHdaLinkInoutClass = 10,
+
+	/* DMIC link input (DSP <-) */
+	kDmicLinkInputClass = 11,
+
+	/* I2S link output (DSP ->) */
+	kI2sLinkOutputClass = 12,
+	/* I2S link input (DSP <-) */
+	kI2sLinkInputClass = 13,
+
+	/* ALH link output, legacy for SNDW (DSP ->) */
+	kALHLinkOutputClass = 16,
+	/* ALH link input, legacy for SNDW (DSP <-) */
+	kALHLinkInputClass = 17,
+
+	/* SNDW link output (DSP ->) */
+	kAlhSndWireStreamLinkOutputClass = 16,
+	/* SNDW link input (DSP <-) */
+	kAlhSndWireStreamLinkInputClass = 17,
```

Comment:
why do we have two definitions for 16 and 17?

Reply:
> why do we have two definitions for 16 and 17?

That was copied from cavs firmware and I don't know the reason why they use the same number. @RanderWang Do you know any reason that we have two definitions for 16 and 17?